File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/options/SetPlayerFrameOptions.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.youtubeplayer.YoutubePlayerHelper;
import io.capawesome.capacitorjs.plugins.youtubeplayer.classes.PlayerFrame;

public class SetPlayerFrameOptions {

    @NonNull
    private final PlayerFrame frame;

    @NonNull
    private final String id;

    public SetPlayerFrameOptions(@NonNull PluginCall call) throws Exception {
        this.frame = PlayerFrame.getFrameFromCall(call);
        this.id = YoutubePlayerHelper.getPlayerIdFromCall(call);
    }

    @NonNull
    public PlayerFrame getFrame() {
        return frame;
    }

    @NonNull
    public String getId() {
        return id;
    }
}

File: packages/youtube-player/ios/Plugin/Classes/Options/SetPlayerFrameOptions.swift
import Capacitor
import Foundation

@objc public class SetPlayerFrameOptions: NSObject {
    let frame: PlayerFrame
    let id: String

    init(_ call: CAPPluginCall) throws {
        self.frame = try PlayerFrame.getFrameFromCall(call)
        self.id = try YoutubePlayerHelper.getPlayerIdFromCall(call)
    }
}
